tests/pset.rs

```diff
@@ -0,0 +1,335 @@
+                0.into(),
+                false.into(),
+            ],
+        );
+        let psbt_base64 = value.as_str().unwrap().to_string();
+
+        assert_eq!(elementsd.expected_next(&psbt_base64), "updater");
+        let psbt_base64 = elementsd.wallet_process_psbt(&psbt_base64);
+        assert_eq!(elementsd.expected_next(&psbt_base64), "finalizer");
+        psbt_rtt(&elementsd, &psbt_base64);
+
+        let tx_hex = elementsd.finalize_psbt(&rtt(&psbt_base64));
+        assert!(elementsd.test_mempool_accept(&tx_hex));
+    }
+
+    #[cfg_attr(feature = "integration", test)]
+    fn tx_pegin() {
+        let (elementsd, bitcoind) = setup(true);
+        let bitcoind = bitcoind.unwrap();
+        let mainchain_address = bitcoind.client.get_new_address(None, None).unwrap();
+        let address_lbtc = elementsd.get_new_address();
+        bitcoind
+            .client
+            .generate_to_address(101, &mainchain_address)
+            .unwrap();
+        let (pegin_address, claim_script) = elementsd.get_pegin_address();
+        let txid = bitcoind
+            .client
+            .send_to_address(
+                &Address::from_str(&pegin_address).unwrap(),
+                Amount::from_sat(100_000_000),
+                None,
+                None,
+                None,
+                None,
+                None,
+                None,
+            )
+            .unwrap();
+        let tx = bitcoind.client.get_raw_transaction(&txid, None).unwrap();
+        let tx_bytes = serialize(&tx);
+        let vout = tx
+            .output
+            .iter()
+            .position(|o| {
+                Address::from_script(&o.script_pubkey, bitcoin::Network::Regtest)
+                    .unwrap()
+                    .to_string()
+                    == pegin_address
+            })
+            .unwrap();
+
+        bitcoind
+            .client
+            .generate_to_address(101, &mainchain_address)
+            .unwrap();
+        let proof = bitcoind.client.get_tx_out_proof(&[txid], None).unwrap();
+        elementsd.generate(2);
+        let inputs = json!([ {"txid":txid, "vout": vout,"pegin_bitcoin_tx": tx_bytes.to_hex(), "pegin_txout_proof": proof.to_hex(), "pegin_claim_script": claim_script } ]);
+        let value = elementsd.call(
+            "createpsbt",
+            &[
+                inputs,
+                json!([
+                    {address_lbtc: "0.9", "blinder_index": 0},
+                    {"fee": "0.1" }
+                ]),
+                0.into(),
+                false.into(),
+            ],
+        );
+        let psbt_base64 = value.as_str().unwrap().to_string();
+        assert_eq!(elementsd.expected_next(&psbt_base64), "updater");
+        let psbt_base64 = elementsd.wallet_process_psbt(&psbt_base64);
+        assert_eq!(elementsd.expected_next(&psbt_base64), "extractor");
+
+        // TODO this fails because elements decodepsbt is not printing TxOut::asset (PSET_IN_WITNESS_UTXO)
```

Comment:
I am using `fn psbt_rtt` to verify compatibility between pset implementations rust-elements and elements, but this requires `decodepsbt` to print everything in the upset and I think it's not the case as written in the above comment, maybe @stevenroose could confirm/deny

I cannot compare binaries because pair fields aren't ordered the same

Reply:
This is tricky. I don't see any solution. Also, in general, I don't like the idea of testing using `decode`RPCs. If I recall correctly even decoderawtransaction does not print all details. But I don't see any other way for doing this :(

Reply:
Implementing `cmp::Eq` on PSET other than being a mess would not accomplish what we are trying to achieve here (PSET implementations compatibility). Suppose for example rust-elements deserialize is skipping a field in the PSET, you wouldn't notice. 

Since encoded binary comparison is excluded because the standard is not enforcing field ordering https://github.com/ElementsProject/elements/issues/1059 I can't think of another way to verify compatibility other than relying on `decodepsbt` printing all fields.

Reply:
Unfortunately, I think we just need to comment this test. Raise an issue in upstream elements to add this, and add a remainder issue in rust-elements to re-add these once upstream elements fixes the bug.